File: client/src/main/java/epam/PingPongClient.java
package epam;

import io.grpc.Channel;

import java.util.logging.Logger;

/**
 * @author Dominik_Janiga
 */
class PingPongClient {

    private static final Logger LOGGER = Logger.getLogger(PingPongClient.class.getName());

    private final PingPongServerGrpc.PingPongServerBlockingStub stub;

    public PingPongClient(Channel channel) {
        this.stub = PingPongServerGrpc.newBlockingStub(channel);
    }

    public void sendMessage() {
        PingRequest message = PingRequest.newBuilder()
                .setMessage("Ping from Java client")
                .build();

        PongResponse pongResponse = this.stub.pingPong(message);
        LOGGER.info("Received response: " + pongResponse.getMessage());
    }
}

File: python-client/epam/ping_pong_client.py
import grpc
import epam.ping_pong_pb2_grpc as pb2_grpc
import epam.ping_pong_pb2 as pb2


def run():
    with grpc.insecure_channel('localhost:50051') as channel:
        stub = pb2_grpc.PingPongServerStub(channel)
        response = stub.pingPong(pb2.PingRequest(message='Ping from Python client'))
    print("Python client received: " + response.message)


if __name__ == '__main__':
    run()
